File: rekognition.py
import boto3

def analyze_emotions(img_bytes, region='eu-central-1'):
    session = boto3.Session(region_name=region)
    client = session.client('rekognition', 
        region_name=region,
        aws_access_key_id='AKIAWCN3YKOJYAHBWZC5',
        aws_secret_access_key='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')

    response = client.detect_faces(Image={'Bytes': img_bytes},
                                   Attributes=['ALL'])

    return str(response['FaceDetails'][0]['Emotions'][0]['Type'])

File: src/pages/services/rekognition.js
import AWS from 'aws-sdk';

AWS.config.update({
    region: process.env.AWS_REGION,
    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
});
const rekognition = new AWS.Rekognition();

export async function analyzeEmotions(base64Image) {
    const imgBytes = Buffer.from(base64Image, 'base64');

    const params = {
        Image: {
            Bytes: imgBytes
        },
        Attributes: ['ALL']
    };

    const data = await rekognition.detectFaces(params).promise();
    const emotionType = data.FaceDetails[0].Emotions[0].Type;

    return { emotionType };
}